### ISSUE TITLE
confusion about the implementation

### ISSUE THREAD

**Issue by ShoufaChen** [facebookresearch/PoseWarper#7]
Thank you for your code. 

 https://github.com/facebookresearch/PoseWarper/blob/master/lib/models/pose_hrnet.py#L307
Here, does the `torch.no_grad()` have any special purposes?   Does this have any effect on the code?

**Comment by gberta** [comment]
I don't think this has any practical effect on the code since we explicitly set which layers need gradient computation during backpropagation, and which don't. We do this because we first train the full pose detection model in a single frame-setting. We then freeze the model, and only train deformable convolution layers responsible for warping pose from one frame to another frame.
